languages/python/sqlalchemy-oso/sqlalchemy_oso/session.py

```diff
@@ -210,12 +213,56 @@ class AuthorizedSession(AuthorizedSessionBase, Session):
     Usually :py:func:`authorized_sessionmaker` is used instead of directly
     instantiating the session.
 
-    NOTE: Unless ``enable_baked_queries=True`` is passed to the constructor,
-          _baked_queries are disabled since the caching mechanism can bypass
-          authorization by using queries from the cache that were previously
-          baked without authorization applied.
+    NOTE: _baked_queries are disabled on SQLAlchemy 1.3 since the caching
+          mechanism can bypass authorization by using queries from the cache
+          that were previously baked without authorization applied. Note that
+          _baked_queries are deprecated as of SQLAlchemy 1.4.
 
-    .. _baked_queries: https://docs.sqlalchemy.org/en/13/orm/extensions/baked.html
+    .. _baked_queries: https://docs.sqlalchemy.org/en/14/orm/extensions/baked.html
     """
 
     pass
+
+
+if AT_LEAST_SQLALCHEMY_VERSION_1_4:
```

Comment:
could we move this up to be an `else` on the `if` above so it's more clear that this is the other implementation?

Reply:
efef45c